docs/the-gazelle-style-guide.md

```diff
@@ -0,0 +1,136 @@
+# The Gazelle's Programming Style Guide
```

Comment:
A thing I could maybe add here is references to examples in our codebase showing how to do things right. I think it's already pretty good though, and I don't know if we have really amazing examples yet?